connection.ts

```diff
@@ -220,8 +220,13 @@ export class Connection {
 
   private async _authMd5(salt: Uint8Array) {
     this.packetWriter.clear();
+
+    if (!this.connParams.password) {
```

Comment:
@denniscollective can you add a test case demonstrating this error?

Reply:
Otherwise LGTM

Reply:
It won't compile without this. If you really want a unit test I can make it, but it would be redundant because that's what the type system is doing in this case.

Reply:
@denniscollective I'm not sure I follow. `_authMd5` is only called if Postgres is configured to require password. So if you don't specify `password` in `connParams` the program should compile just fine, because it's a runtime condition if it's required or not.

Reply:
a unit test in this case is unnecessary (IMO) because if you remove that ```if (!this.connParams.password) ``` guard clause the system will no longer type check. This is an issue that was surfaced by enabling strict mode on typescript, which I think demonstrates the value of this pull request.

I'm burnt out on the back-and-forth in this pull request, and requests to fix code that I had nothing to do with and didn't touch, if you want to merge this request, great! If you want a test case about the issue that strict mode surfaced, I'd suggest making an issue, tagging it easy-to-fix or something like that, and direct people who are excited about contributing to do it.

I have already have a full time job and am no longer excited about contributing to this project.

Thanks again for the library, it is fun to play around with, and good luck!

Reply:
@denniscollective I'm sorry I just figured out, that tests with password are commented out (`tests/client.ts`) - I need to fix that. Please don't take any hard feelings, this PR is very valuable and I'm grateful for your contribution.